mlflow/types/utils.py

```diff
@@ -103,46 +166,24 @@ def _infer_datatype(data) -> DataType:
     raise MlflowException.invalid_parameter_value("Data is not one of the supported DataType")
 
 
-def _infer_colspec_type(data: Any) -> Union[DataType, Array, Object]:
+def _validate_inferred_type(dtype: Any):
     """
-    Infer an MLflow Colspec type from the dataset.
-
-    :param data: data to infer from.
-    :return: Object
+    Validate inferred type for a column. Currently it's trivial check just to avoid unexpected
```

Comment:
note: This might be redundant as we have type check in `Array`, `Object` constructor, but taking safer path considering how bad it is for users to get a broken schema silently. Happy to see other thoughts.